### PR TITLE
oidc: add code challenge supported methods to the discovery doc

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: hectorj2f <hectorf@vmware.com>

This PR includes the `code_challenge_methods_supported` key, as specified by https://datatracker.ietf.org/doc/html/rfc8414#section-2. If this key is omitted in the document, that'll mean the authorization server does not support PKCE. However the Supervisor does actually support and require PKCE, therefore we should add that key in the discovery document.

Fixes https://github.com/vmware-tanzu/pinniped/issues/1107

**Release note**:

```release-note
oidc: add code challenge supported methods to the discovery document.
```

### CODE REVIEW

File: internal/oidc/discovery/discovery_handler.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type Metadata struct {
 	TokenEndpointAuthMethodsSupported []string `json:"token_endpoint_auth_methods_supported"`
 	ScopesSupported                   []string `json:"scopes_supported"`
 	ClaimsSupported                   []string `json:"claims_supported"`
+	CodeChallengeMethosSupport        []string `json:"code_challenge_methods_supported"`
```
Comment (cfryanr) [inline internal/oidc/discovery/discovery_handler.go:40]: Could we change the name of this field to `CodeChallengeMethodsSupported`? Seems like maybe this was a typo.

Also, I think this test expectation in the integration tests will need to be updated to include the new field... https://github.com/vmware-tanzu/pinniped/blob/53597bb82492f134c0b4fb01d4173a19922a7457/test/integration/supervisor_discovery_test.go#L499-L512

**Review by cfryanr — CHANGES_REQUESTED** [review]
Thanks for the PR @hectorj2f! I added a comment with a couple of suggestions.

File: internal/oidc/discovery/discovery_handler.go (hunk 1 above)
Comment (hectorj2f) [reply internal/oidc/discovery/discovery_handler.go]: You are right! I fixed the typo, and added the field to the new test.

**Review by cfryanr — APPROVED** [review]
Approved to run through CI.

**Pull request merged**